server/libs/domain/src/partner/partner.service.ts

```diff
@@ -32,7 +32,9 @@ export class PartnerService {
   async getAll(authUser: AuthUserDto, direction: PartnerDirection): Promise<UserResponseDto[]> {
     const partners = await this.repository.getAll(authUser.id);
     const key = direction === PartnerDirection.SharedBy ? 'sharedById' : 'sharedWithId';
-    return partners.filter((partner) => partner[key] === authUser.id).map((partner) => this.map(partner, direction));
+    return partners
+      .filter((partner) => partner.sharedBy && partner[key] === authUser.id)
```

Comment:
Can you instead add a new filter clause that checks both `sharedBy` and `sharedWith` are not null? We should make sure it is fixed in both directions and I think one filter (maybe with a comment, because it is not obvious) specifically for the deleted users case and another one for the direction would be more clear.